xilem/src/view/zstack.rs

```diff
@@ -0,0 +1,378 @@
+            if self.alignment != prev.alignment {
+                widget::ZStack::update_child_alignment(
+                    &mut element.parent,
+                    element.idx,
+                    self.alignment,
+                );
+            }
+            let mut child = widget::ZStack::child_mut(&mut element.parent, element.idx)
+                .expect("ZStackWrapper always has a widget child");
+            self.view
+                .rebuild(&prev.view, view_state, ctx, child.downcast());
+        }
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        mut element: Mut<Self::Element>,
+    ) {
+        let mut child = widget::ZStack::child_mut(&mut element.parent, element.idx)
+            .expect("ZStackWrapper always has a widget child");
+        self.view.teardown(view_state, ctx, child.downcast());
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[ViewId],
+        message: DynMessage,
+        app_state: &mut State,
+    ) -> MessageResult<Action> {
+        self.view.message(view_state, id_path, message, app_state)
+    }
+}
+
+// --- MARK: ZStackElement ---
+
+/// A struct implementing [`ViewElement`] for a `ZStack`.
+pub struct ZStackElement {
+    widget: Pod<Box<dyn Widget>>,
+    alignment: ChildAlignment,
+}
+
+/// A mutable version of `ZStackElement`.
+pub struct ZStackElementMut<'w> {
+    parent: WidgetMut<'w, widget::ZStack>,
+    idx: usize,
+}
+
+impl ZStackElement {
+    fn new(widget: Pod<Box<dyn Widget>>, alignment: ChildAlignment) -> Self {
+        Self { widget, alignment }
+    }
+}
+
+impl ViewElement for ZStackElement {
+    type Mut<'a> = ZStackElementMut<'a>;
+}
+
+impl SuperElement<Self, ViewCtx> for ZStackElement {
+    fn upcast(_ctx: &mut ViewCtx, child: Self) -> Self {
+        child
+    }
+
+    fn with_downcast_val<R>(
+        mut this: Mut<Self>,
+        f: impl FnOnce(Mut<Self>) -> R,
+    ) -> (Self::Mut<'_>, R) {
+        let r = {
+            let parent = this.parent.reborrow_mut();
+            let reborrow = ZStackElementMut {
+                idx: this.idx,
+                parent,
+            };
+            f(reborrow)
+        };
+        (this, r)
+    }
+}
+
+impl<W: Widget> SuperElement<Pod<W>, ViewCtx> for ZStackElement {
+    fn upcast(ctx: &mut ViewCtx, child: Pod<W>) -> Self {
+        Self::new(ctx.boxed_pod(child), ChildAlignment::ParentAligned)
+    }
+
+    fn with_downcast_val<R>(
+        mut this: Mut<Self>,
+        f: impl FnOnce(Mut<Pod<W>>) -> R,
+    ) -> (Self::Mut<'_>, R) {
+        let ret = {
+            let mut child = widget::ZStack::child_mut(&mut this.parent, this.idx)
+                .expect("This is supposed to be a widget");
+            let downcast = child.downcast();
+            f(downcast)
+        };
+
+        (this, ret)
+    }
+}
+
+// MARK: Sequence
+
+/// A trait implementing `ViewSequence` for `ZStackElements`.
```

Comment:
Nit: I think using the plural for `ZStackElement` may suggest that there's a type `ZStackElements`, so maybe rather `ZStackElement`s or 
```
[`ZStackElement`]s
```
to directly link to the type.

(same for the other occurences of this here)

Reply:
Makes sense, I fixed it in the latest commit